graphHttpServer/http_server.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2012-2013, Aptarism SA.
+//
+// All rights reserved.
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright
+//   notice, this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright
+//   notice, this list of conditions and the following disclaimer in the
+//   documentation and/or other materials provided with the distribution.
+// * Neither the name of the University of California, Berkeley nor the
+//   names of its contributors may be used to endorse or promote products
+//   derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE REGENTS AND CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#include "http_server.h"
+
+#include <string.h>
+#include <iostream>
+
+//#include <base/string.h>
+
+#include "civetweb/include/CivetServer.h"
+
+
+class HttpServerCivetHandler: public CivetHandler {
+  public:
+    HttpServerCivetHandler(HttpServer* server) : server_(server) { }
+
+    bool handleGet(CivetServer *server, struct mg_connection *conn) {
+        const struct mg_request_info *request_info = mg_get_request_info(conn);
+          //printf("New request: uri: %s query_string: %s\n",
+          //       request_info->uri, request_info->query_string);
+          // TODO: make_shared
+          HttpReply * reply = new HttpReply(conn);
```

Comment:
This could be modernized and pass a std::unique_ptr<HttpReply>, move it to onNewRequest() and there's no need for explicit delete

Reply:
done